Library/Homebrew/hardware.rb

```diff
@@ -39,13 +71,17 @@ def family
       end
 
       def cores
-        1
+        return @cores if @cores
+        @cores = Utils.popen_read("getconf", "_NPROCESSORS_ONLN").chomp.to_i
+        @cores = 1 unless $CHILD_STATUS.success?
+        @cores
       end
 
       def bits
-        case RUBY_PLATFORM
-        when /x86_64/, /ppc64/ then 64
-        when /i\d86/, /ppc/ then 32
+        @bits ||= case RUBY_PLATFORM
+        when /x86_64/, /ppc64/, /aarch64/, /arm64/ then 64
+        when /i\d86/, /ppc/, /arm/ then 32
+        else Utils.popen_read("getconf", "LONG_BIT").chomp.to_i
```

Comment:
Let's avoid the `else` here unless you have a platform that you know it will run on. Alternatively, `else 64` because we default to 64-bit everywhere.

Reply:
Done.